genai-perf/genai_perf/inputs/inputs.py

```diff
@@ -81,6 +76,18 @@ def _write_json_to_file(self, json_in_pa_format: Dict) -> None:
         with open(str(filename), "w") as f:
             f.write(json.dumps(json_in_pa_format, indent=2))
 
+    def _check_for_supported_input_type(self) -> None:
```

Comment:
This is a nice addition. Long-term, it'd be good to move toward a more generic approach toward supported/unsupported types. We could even move it to a file that's easy for users to update when adding a new task.

e.g.
``` python
unsupported_combinations = {
            OutputFormat.OPENAI_EMBEDDINGS: [PromptSource.DATASET, PromptSource.SYNTHETIC],
            OutputFormat.RANKINGS: [PromptSource.DATASET, PromptSource.SYNTHETIC],
        }
if input_type in unsupported_combinations.get(output_format, []):
            raise GenAIPerfException(
                f"{dataset_name} does not have a corresponding URL in the dataset_url_map."
                f"{output_format.to_lowercase()} does not support input type `{input_type.to_lowercase()}`"
            )
```

Reply:
Nice suggestion! Where do you think is a good place to add this? Atm, I can't think of other locations in mind other than `inputs.py`.

Reply:
Thanks! A generic format like this could even go within this function. Alternatively, it could be supported combinations, though I think that list would be much longer.

Longer-term, we'd want to centralize it somewhere. We could drop it in one of the constants files, but I think we'll ultimately want to think through the workflow of adding a new task and go from there.

One idea is the converter could be responsible for doing the check, since it's tied to output format and that means there's one fewer file the user has to touch to add a task. Lots of options though! Just seeding some ideas. This doesn't have to be resolved in this PR.

Reply:
Sounds good. Created TPA-425.